hydro_deploy/core/src/ssh.rs

```diff
@@ -438,14 +436,34 @@ impl<T: LaunchedSshHost> LaunchedHost for T {
                 }
                 // Launch with perf if specified, also copy local binary to expected place for perf report to work
                 if let Some(TracingOptions { frequency, .. }) = tracing.clone() {
+
+                    // Install perf
+                    let mut perf_install_channel = create_channel(&session).await?;
+                    perf_install_channel
```

Comment:
If we can, let's do a bit of tiding up here. We can add to `trait LaunchedSshHost` an `install_perf_script() -> Option<String>`. We then add a config option to GPC hosts `perf_script: Option<String>` which gets passed down to the launched host. And here, we only do this logic if the option is `Some`. Just so that the user can easily configure the script, since it will easily vary across Linux distros.

Reply:
In that way, the examples won't change except instead of calling `startup_script` we call `install_perf_script`.

Reply:
Thinking more, we could make this even more general and call it `pre_launch_script`.